File: Linear Data Structure/Array/Java/ArrayDeclarationWays.java
public class ArrayDeclarationWays {
    public static void main(String[] args) {
        // Declaration with initialization
        int[] a = {1, 2, 3};

        // Declaration then initialization
        int[] b;
        b = new int[]{4, 5, 6};

        // Declaring with size, then assigning values
        int[] c = new int[3];
        c[0] = 7;
        c[1] = 8;
        c[2] = 9;

        // Printing all arrays
        System.out.println("Array a:");
        for (int val : a) System.out.print(val + " ");
        System.out.println("\nArray b:");
        for (int val : b) System.out.print(val + " ");
        System.out.println("\nArray c:");
        for (int val : c) System.out.print(val + " ");
    }
}

File: Linear Data Structure/Array/C/ArrayDeclarationWays.c
#include <stdio.h>

int main() {
    // Way 1: Declaration with size and initialization
    int arr1[5] = {1, 2, 3, 4, 5};

    // Way 2: Declaration without size
    int arr2[] = {6, 7, 8};

    // Way 3: Only declaration, values assigned later
    int arr3[3];
    arr3[0] = 9;
    arr3[1] = 10;
    arr3[2] = 11;

    // Print all arrays
    printf("arr1: ");
    for (int i = 0; i < 5; i++) printf("%d ", arr1[i]);

    printf("\narr2: ");
    for (int i = 0; i < 3; i++) printf("%d ", arr2[i]);

    printf("\narr3: ");
    for (int i = 0; i < 3; i++) printf("%d ", arr3[i]);

    return 0;
}
